### DISCUSSION TITLE
What is the DefaultModel in appsettings.json?

### DISCUSSION THREAD

**Discussion by djtyambao** [marcominerva/DatabaseGPT#75] [Q&A] [answered]
Hi, I tried to run the application, but I am receiving an error message that says, "The model my-model does not exist or you do not have access to it."

Currently, the appsettings.json default value is set to -> "DefaultModel": "my-model",
What setting do I need to assign in the DefaultModel? I am using the Northwind Database as test data.
I appreciate your help.

Thank you,

Jake Crisostomo

**Comment by marcominerva** [comment]
You need to specify the configuration of your OpenAI model, like described at https://github.com/marcominerva/ChatGptNet?tab=readme-ov-file#configuration.

**Reply by djtyambao** [reply]
Thank you for the quick response.

I am getting undefined coming from the OpenAI. Is this the correct parameter in the appsettings?
"DefaultModel": "o1-preview",

**Reply by marcominerva** [reply]
It depends on your configuration and to which models you have access.

**Reply by djtyambao** [reply]
> It depends on your configuration and to which models you have access.

I am trying to mimic your sample, I would like to query on the Northwind database just like in your video. I tried all the models but I am getting "Undefined" error from the Bot. It's my first time doing this OpenAI. Anyways, is this something to do with the API key restriction? Thank you in advance

**Reply by marcominerva** [reply, marked as answer]
You need to set the API Key and the model name using the name that is provided by OpenAI.

**Reply by djtyambao** [reply]
I generated a key. It already exceeded the quota. I ran the console and saw the error. Thank you
